utils/analytics/useAnalytics.ts

```diff
@@ -61,3 +71,71 @@ export const useAnalytics = () => {
 		page,
 	} as Analytics;
 };
+
+/**
+ * Custom hook that sends a page view event to the analytics services only once.
+ *
+ * Also handles rerender logic for GDPR consent when using Google Analytics.
+ *
+ * Flow for non-Google Analytics and Google Analytics + consent already given:
+ *
+ * 1. User visits page
+ * 2. Page view event is sent to all analytics services
+ *
+ * Flow for Google Analytics + no consent yet given:
+ *
+ * 1. User visits page
+ * 2. Page view event is sent to our analytics service
+ * 3. User clicks "Accept" on the GDPR banner
+ * 4. Page view is sent to Google Analytics.
+ *
+ * Flow for Google Analytics + consent refused:
+ *
+ * 1. User visits page
+ * 2. Page view event is sent to only our analytics service
+ *
+ * @param payload - The payload containing the page view data or a function that returns the
+ *   payload.
+ * @param gdprConsent - Optional boolean indicating whether GDPR consent has been given.
+ */
+export const usePageOnce = (
+	payload: PageViewPayload | (() => PageViewPayload | null),
```

Comment:
What's the reasoning for having `payload` be either the payload or a function that returns a payload (invoked immediately with no arguments)? It appears to be used only in the former case.